apis/projectcontour/v1alpha1/extensionservice.go

```diff
@@ -88,11 +88,11 @@ type ExtensionServiceSpec struct {
 
 	// This field sets the version of the GRPC protocol that Envoy uses to
 	// send requests to the extension service. Since Contour always uses the
-	// v2 Envoy API, this is currently fixed at "v2". However, other
+	// v3 Envoy API, this is currently fixed at "v3". However, other
 	// protocol options will be available in future.
 	//
 	// +optional
-	// +kubebuilder:validation:Enum=v2
+	// +kubebuilder:validation:Enum=v3
```

Comment:
```suggestion
	// +kubebuilder:validation:Enum=v2,v3
```

Either v2 or v3 is OK since it's whatever the service happened to implement. Might be out of the control of the operator.

Reply:
Is v2 still going to be a valid option when Envoy drops xDS v2 support? I was assuming it would drop v2 support for the ext auth API as well, so we wouldn't want to allow it to be configured.

Reply:
@jpeach @youngnick @stevesloka I need some further input on this, so we can get this PR in before 1.11.

Reply:
I think it's fine to only support v3. v2 will be dropped in a month, also, the field isn't even implemented in the code, it's only in the API, so even if you configure it, it's not going to change any logic.

Reply:
We should just drop the field. How far off is v4 and the api is alpha, so do we even need to carry it forward?

Reply:
According to https://docs.google.com/document/d/1afQ9wthJxofwOMCAle8qF-ckEfNWUuWtxbmdhxji0sI/edit# (still in review/not sure the status of the proposal) it seems like major version bumps/deprecations aren't going to be yearly anymore, `v4alpha` changes have started (https://github.com/envoyproxy/envoy/tree/7d348d1b65d3c5ae58d0259e397fbd3df8686f11/api/envoy/admin/v4alpha) but I would guess there is time

+1 to drop the field and bring it back for v3->v4+ if needed

Reply:
This API is alpha, so we can just drop the field. I agree that we should do that - I think we need to consider where we put settings that are for a particular type of ExtensionService (ie one that needs gRPC, or one that connects to a tracing service and needs properties set.) I don't think top-level fields are right for that, so we probably will need to remove this anyway. Let's just remove it now, and come back and add things if we need to later, in a more sustainable way.

Reply:
I *believe* that we'd need to bump this API to `v1alpha2` in order to drop a field, per [rule 1](https://kubernetes.io/docs/reference/using-api/deprecation-policy/#deprecating-parts-of-the-api).

I can look at doing that, but maybe a simpler solution for now is to leave it as I currently have it in this PR - the field still exists, but it's not being used, and Envoy is being programmed to use `v3`. Then when we're ready to make a rev to this API, we can drop the field if we still think that makes sense?

Reply:
I don't think a version rev at this point is needed. My only thing is that it's not implemented so what are we breaking? =)

We can just move on from here, but need to find a place to put this type of discussion for the future without TODO's all over the place that never get used. Should we have a spot in the wiki? Or an issue?